### PR TITLE
Cells revision

### PR DESCRIPTION
**Pull request by AnnBibyk**

### CODE REVIEW

File: Sources/Controllers/Cells/OAIconTextTableViewCell.h [hunk 1]
```diff
@@ -18,6 +18,8 @@
 @property (nonatomic) IBOutlet NSLayoutConstraint *textLeftMarginNoImage;
 @property (nonatomic) IBOutlet NSLayoutConstraint *textRightMargin;
 @property (nonatomic) IBOutlet NSLayoutConstraint *textRightMarginNoImage;
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *iconViewHeightConstraint;
```
Comment (max-klaus) [inline Sources/Controllers/Cells/OAIconTextTableViewCell.h:21]: weak?!

File: Sources/Controllers/Cells/OAWaypointHeaderCell.h [hunk 2]
```diff
@@ -15,6 +15,9 @@
 @property (weak, nonatomic) IBOutlet UISwitch *switchView;
 @property (weak, nonatomic) IBOutlet UIButton *imageButton;
 @property (weak, nonatomic) IBOutlet UIButton *textButton;
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *leftTitleMarginNoProgress;
```
Comment (max-klaus) [inline Sources/Controllers/Cells/OAWaypointHeaderCell.h:18]: weak!

File: Sources/Cards/OAImageCard.m [hunk 3]
```diff
@@ -115,7 +115,6 @@ - (void) build:(UICollectionViewCell *) cell
                 [imageCell.loadingIndicatorView stopAnimating];
             }
         }
-        imageCell.usernameLabel.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
```
Comment (PaulStets) [inline Sources/Cards/OAImageCard.m:118]: Why was it removed?

File: Sources/Controllers/2.0/OAEditColorViewController.m [hunk 4]
```diff
@@ -86,27 +88,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
         
         OAFavoriteColor *favCol = [OADefaultFavorite builtinColors][indexPath.row];
         [cell.textView setText:favCol.name];
-        [cell.titleIcon setImage:favCol.icon];
-        [cell.iconView setImage:nil];
-        
-        if (indexPath.row == self.colorIndex)
-            [cell.iconView setImage:[UIImage imageNamed:@"menu_cell_selected"]];
+        [cell.iconView setImage:favCol.icon];
+        //[cell.iconView setImage:nil];
```
Comment (PaulStets) [inline Sources/Controllers/2.0/OAEditColorViewController.m:104]: Remove comment

File: Sources/Controllers/2.0/OAEditColorViewController.m [hunk 5]
```diff
@@ -86,27 +88,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
 }
 
+- (CGFloat) tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    return UITableViewAutomaticDimension;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    return UITableViewAutomaticDimension;
```
Comment (PaulStets) [inline Sources/Controllers/2.0/OAEditColorViewController.m:119]: estimatedHeightForRowAtIndexPath should not be automatic + both this methods are redundant. You should set the values in the xib and remove them

File: Sources/Controllers/2.0/OAMapSettingsContourLinesScreen.mm [hunk 6]
```diff
@@ -640,25 +639,22 @@ - (UITableViewCell *) tableView:(UITableView *)tableView cellForRowAtIndexPath:(
...
-            [cell.buttonView setTitle:item[@"title"] forState:UIControlStateNormal];
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:@"OATitleRightIconCell" owner:self options:nil];
+            cell = (OATitleRightIconCell *)[nib objectAtIndex:0];
+            cell.iconView.image = [[UIImage imageNamed:item[@"img"]] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+            cell.titleView.text = item[@"title"];
+            cell.titleView.font = [UIFont systemFontOfSize:17. weight:UIFontWeightSemibold];
+            cell.titleView.textColor = UIColorFromRGB(color_primary_purple);
             cell.iconView.tintColor = UIColorFromRGB(color_primary_purple);
         }
-        [cell.buttonView removeTarget:self action:NULL forControlEvents:UIControlEventTouchUpInside];
```
Comment (PaulStets) [inline Sources/Controllers/2.0/OAMapSettingsContourLinesScreen.mm:659]: Why was target removed?

File: Sources/Controllers/Cells/OAPublicTransportShieldCell.h [hunk 7]
```diff
@@ -23,7 +23,4 @@
 -(void) setData:(SHARED_PTR<TransportRouteResult>)data;
 -(void) needsSafeAreaInsets:(BOOL)needsInsets;
 
-+ (CGFloat) getCellHeight:(CGFloat)width route:(SHARED_PTR<TransportRouteResult>)route;
```
Comment (PaulStets) [inline Sources/Controllers/Cells/OAPublicTransportShieldCell.h:26]: Transport shields are broken, revert and fix correctly!

Comment (PaulStets) [reply Sources/Controllers/Cells/OAPublicTransportShieldCell.h:26]: Edit: perhaps simply setting the background color and removing the separator will fix it, but still you need to check such things!

File: Sources/Controllers/Cells/OAPublicTransportShieldCell.mm [hunk 8]
```diff
@@ -315,47 +328,6 @@ - (UIImageView *) createArrowImageView
     return [NSArray arrayWithArray:titles];
 }
 
-+ (CGFloat) getCellHeight:(CGFloat)width route:(SHARED_PTR<TransportRouteResult>)route
```
Comment (PaulStets) [inline Sources/Controllers/Cells/OAPublicTransportShieldCell.mm:318]: revert

**Pull request merged**